sql/postgresql/PostgreSQLParser.g4

```diff
@@ -2865,55 +2779,55 @@ not_la: NOT;
        | b_expr qual_op
 
        //S ISNULL NOTNULL	 	IS TRUE, IS FALSE, IS NULL, IS DISTINCT FROM, etc
-       | b_expr IS DISTINCT FROM b_expr
-       | b_expr IS NOT DISTINCT FROM b_expr
-       | b_expr IS OF OPEN_PAREN type_list CLOSE_PAREN
-       | b_expr IS NOT OF OPEN_PAREN type_list CLOSE_PAREN
-       | b_expr IS DOCUMENT_P
-       | b_expr IS NOT DOCUMENT_P
-;
- c_expr: columnref # c_expr_expr
+       | b_expr IS NOT?
+                (
+                    DISTINCT FROM b_expr
+                    |OF OPEN_PAREN type_list CLOSE_PAREN
+                    |DOCUMENT_P
+                )
+;
+ c_expr: PARAM opt_indirection # c_expr_expr
+       | EXISTS select_with_parens # c_expr_exists
+       | ARRAY (select_with_parens|array_expr) # c_expr_expr
+       | GROUPING OPEN_PAREN expr_list CLOSE_PAREN # c_expr_expr
+       | columnref # c_expr_expr
```

Comment:
If we match `columnref` before `func_expr`, won't `f(1)` matches `f` as columnref first and remains `(1)` as illegal?

cc @OleksiiKovalov @KvanTTT

Reply:
I know that ANTLR 4 uses LL(*) to de-ambiguous now. I'm using a parser combinator (nom) so I may need to peek and look ahead by myself. Or properly ordering the matching order.